File: backend/src/main/java/com/ll/hotel/domain/hotel/hotel/dto/PostHotelRequest.java
package com.ll.hotel.domain.hotel.hotel.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import java.time.LocalTime;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Objects;
import java.util.Set;
import org.hibernate.validator.constraints.Length;

public record PostHotelRequest(
        @NotBlank @Length(min = 2, max = 30)
        String hotelName,

        @Email @NotBlank
        String hotelEmail,

        @NotBlank @Pattern(regexp = "^01[0-9]-\\d{4}-\\d{4}$", message = "Invalid phone number format")
        String hotelPhoneNumber,

        @NotBlank
        String streetAddress,

        @NotNull
        Integer zipCode,

        @NotNull @Min(value = 0) @Max(value = 5)
        Integer hotelGrade,

        @NotNull
        @JsonFormat(pattern = "HH:mm")
        LocalTime checkInTime,

        @NotNull
        @JsonFormat(pattern = "HH:mm")
        LocalTime checkOutTime,

        @NotBlank
        String hotelExplainContent,

        List<String> imageExtensions,

        Set<String> hotelOptions
) {
    public PostHotelRequest {
        imageExtensions = Objects.requireNonNullElse(imageExtensions, new ArrayList<>());
        hotelOptions = Objects.requireNonNullElse(hotelOptions, new HashSet<>());
    }
}

File: backend/src/main/java/com/ll/hotel/domain/hotel/hotel/dto/PostHotelRequest.kt
package com.ll.hotel.domain.hotel.hotel.dto

import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.validation.constraints.*
import org.hibernate.validator.constraints.Length
import java.time.LocalTime

data class PostHotelRequest(
    @field:NotBlank
    @field:Length(min = 2, max = 30)
    val hotelName: String,

    @field:Email
    @field:NotBlank
    val hotelEmail: String,

    @field:NotBlank
    @field:Pattern(regexp = "^01[0-9]-\\d{4}-\\d{4}$")
    val hotelPhoneNumber: String,

    @field:NotBlank
    val streetAddress: String,

    val zipCode: Int,

    @field:Min(0)
    @field:Max(5)
    val hotelGrade: Int,

    @field:JsonFormat(pattern = "HH:mm")
    val checkInTime: LocalTime,

    @field:JsonFormat(pattern = "HH:mm")
    val checkOutTime: LocalTime,

    @field:NotBlank
    val hotelExplainContent: String,

    val imageExtensions: List<String> = emptyList(),

    val hotelOptions: Set<String> = emptySet()
)